jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/nio/file/FileSystemsTest.java

```diff
@@ -0,0 +1,202 @@
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystemAlreadyExistsException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.ProviderNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import libcore.io.Streams;
+
+/* J2ObjC removed: unsupported
+import dalvik.system.PathClassLoader;
+import junitparams.JUnitParamsRunner;
+ */
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+
+/* J2ObjC removed: junitparams unsupported
+@RunWith(JUnitParamsRunner.class)
+ */
+public class FileSystemsTest {
+
+    @Rule
+    public FilesSetup filesSetup = new FilesSetup();
+
+    @Test
+    public void test_getDefault() {
+        FileSystem fs = FileSystems.getDefault();
+        assertNotNull(fs.provider());
+    }
+
+    @Test
+    public void test_getFileSystem() {
+        Path testPath = Paths.get("/");
+        FileSystem fs = FileSystems.getFileSystem(testPath.toUri());
+        assertNotNull(fs.provider());
+
+        try {
+            FileSystems.getFileSystem(null);
+            fail();
+        } catch (NullPointerException expected) {}
+    }
+
+    @Test
+    public void test_newFileSystem$URI$Map() throws IOException {
+        Path testPath = Paths.get("/");
+        Map<String, String> stubEnv = new HashMap<>();
+        try {
+            FileSystems.newFileSystem(testPath.toUri(), stubEnv);
+            fail();
+        } catch (FileSystemAlreadyExistsException expected) {}
+
+        try {
+            FileSystems.newFileSystem(null, stubEnv);
+            fail();
+        } catch (NullPointerException expected) {}
+
+        try {
+            FileSystems.newFileSystem(testPath, null);
+            fail();
+        } catch (ProviderNotFoundException expected) {}
+    }
+
+    @Test
+    public void test_newFileSystem$URI$Map$ClassLoader() throws Exception {
+        Path testPath = Paths.get("/");
+        Map<String, String> stubEnv = new HashMap<>();
+        try {
+            FileSystems.newFileSystem(testPath.toUri(), stubEnv, getClass().getClassLoader());
+            fail();
+        } catch (FileSystemAlreadyExistsException expected) {}
+
+        try {
+            FileSystems.newFileSystem(null, stubEnv,
+                    Thread.currentThread().getContextClassLoader());
+            fail();
+        } catch (NullPointerException expected) {}
+
+        try {
+            FileSystems.newFileSystem(testPath.toUri(), null,
+                    Thread.currentThread().getContextClassLoader());
+            fail();
+        } catch (FileSystemAlreadyExistsException expected) {}
+
+        try {
+            FileSystems.newFileSystem(testPath.toUri(), stubEnv, null);
+            fail();
+        } catch (FileSystemAlreadyExistsException expected) {}
+    }
+
+    /* TODO(amisail): try to port PathClassLoader
```

Comment:
PathClassLoader shouldn't be ported since it extends BasicDexClassLoader, and "dex" files are unique to Android.

Reply:
Fixed